modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -624,7 +651,24 @@ void ONNXCompiled::Run(const std::vector<cv::Mat>& ins,
                                              ins[i]));
     }
 
-    if (!is_dynamic) {
+    // Order of names for Run
+    auto input_names = params.input_names;
+    if (!params.const_inputs.empty()) {
+        for (auto &&c_in_pair : params.const_inputs) {
+            const auto idx = getIdxByName(in_tensor_info, c_in_pair.first);
+            // Puts const input names in sequence for Run
+            // Ort can math input tensors to CNN inputs by names
+            input_names.emplace_back(c_in_pair.first);
+            in_tensors.emplace_back(createTensor(this_memory_info,
+                                                 in_tensor_info[idx],
+                                                 c_in_pair.second.first));
+        }
+        const auto num_net_in = this_session.GetInputCount();
+        GAPI_Assert(input_names.size() == num_net_in);
```

Comment:
What if there would be similar names in `params.input_names` and `params.const_inputs`?

Reply:
Fixed.